src/plugins/intel_cpu/src/nodes/unique.cpp

```diff
@@ -318,178 +315,171 @@ void Unique::slicedTensorExec() {
-                    ocSrc = occurTmpPtr[0];
+                    std::swap(occurN1, occurN2);
                 }
-                for (size_t k = 0; k < uniqueLen; k++) {
-                    if (mPos == moveTo[mPos]) {
-                        mPos = moveTo[mPos + 1];
-                        continue;
-                    }
-
-                    if (definedOutputs[FIRST_UNIQUE_IDX]) {
-                        auto& fSrc = k % 2 == 0 ? firstSrc : firstDst;
-                        auto& fDst = k % 2 == 0 ? firstDst : firstSrc;
-                        fDst = firstTmpPtr[mPos];
-                        firstTmpPtr[mPos] = fSrc;
-                    }
-                    if (definedOutputs[OCCURRENCES_NUM]) {
-                        auto& oSrc = k % 2 == 0 ? ocSrc : ocDst;
-                        auto& oDst = k % 2 == 0 ? ocDst : ocSrc;
-                        oDst = occurTmpPtr[mPos];
-                        occurTmpPtr[mPos] = oSrc;
-                    }
-
-                    mPos = moveTo[mPos];
+                if (definedOutputs[INPUT_TO_UNIQ_IDX]) {
+                    std::swap(inToOut1, inToOut2);
                 }
             }
         }
 
+        if (definedOutputs[UNIQUE_DATA] && dst1 != dstDataPtr) {
+            memcpy(dstDataPtr, dst1, dstUniDataLen * sizeof(T));
+        }
+        if (definedOutputs[FIRST_UNIQUE_IDX] && first2 != firstPtr) {
+            memcpy(firstPtr, first2, uniqueLenIB);
+        }
+        if (definedOutputs[INPUT_TO_UNIQ_IDX] && inToOut2 != inToOutPtr) {
+            memcpy(inToOutPtr, inToOut2, axisDim * sizeof(int));
+        }
+        if (definedOutputs[OCCURRENCES_NUM] && occurN2 != occurNPtr) {
+            memcpy(occurNPtr, occurN2, uniqueLenIB);
+        }
+    } else {
+        if (definedOutputs[FIRST_UNIQUE_IDX]) {
+            memcpy(firstPtr, firstUniTmp.data(), uniqueLenIB);
+        }
         if (definedOutputs[INPUT_TO_UNIQ_IDX]) {
-            for (size_t b1 = 0; b1 < cmpBlNum; b1++) {
-                auto first1 = srcDataPtr + b1 * elPerPart;
-                auto last1 = srcDataPtr + (b1 + 1) * elPerPart;
-                bool equal = true;
-                for (size_t b2 = 0; b2 < uniqueLen; b2++) {
-                    auto first2 = uniDataTmpPtr + b2 * elPerPart;
-                    equal = true;
-                    for (int p = 0; p < partsInBl; p++) {
-                        equal = std::equal(first1, last1, first2);
-                        if (!equal) {
-                            break;
-                        }
-                        first2 += dstPrtStep;
-                    }
-                    if (equal) {
-                        inToOutTmpPtr[b1] = b2;
-                    }
-                }
-            }
+            memcpy(inToOutPtr, inToOutTmp.data(), axisDim * sizeof(int));
+        }
+        if (definedOutputs[OCCURRENCES_NUM]) {
+            memcpy(occurNPtr, occurTmp.data(), uniqueLenIB);
```

Comment:
Two questions here.
1. Is not it better to use `cpu_memcpy` to avoid some warnings from MSVC?
2. Does it make sense to use parallel copy here?

Reply:
Done